src/Agent.js

```diff
@@ -137,6 +145,9 @@ export default class Agent {
 		if (query) {
 			req.query(query);
 		}
+		if (timeout!==undefined) {
+			req.timeout(timeout);
```

Comment:
What is the user-facing behavior of a timeout? This is an public interface change. An opt-in one but a change nevertheless. How would a user program deal with a timeout vs an HTTP error?